File: answers/answers-rb/grid.rb
path_relative_base = '../../questions/'
path =
  File.join(File.dirname(__FILE__), path_relative_base, 'grid/directions.txt')
f = File.readlines path

@x = 0
@y = 0

f.each do |line|
  direction, magnitude = line.split(/\s+/)
  case direction
  when 'NORTH'
    @y += magnitude.to_i
  when 'SOUTH'
    @y -= magnitude.to_i
  when 'EAST'
    @x += magnitude.to_i
  when 'WEST'
    @x -= magnitude.to_i
  else
    puts "something is wrong"
  end
end

puts "End at x: #{@x}, y: #{@y}"

# End at x: 6666, y: -5555

File: answers/answers-js/grid.js
const fs = require("fs")
data = fs.readFileSync("../../questions/grid/directions.txt","ascii")
const lines = data.split("\n")
let x = 0, y = 0
for (let line of lines) {
  let direction = line.substr(0,1)
  let amount = parseInt(line.split(" ")[1])
  switch (direction) {
    case "N":
      y+=amount; break;
    case "S":
      y-=amount; break;
    case "E":
      x+=amount; break;
    case "W":
      x-=amount; break;
  }
}
console.log(x, y)